src/content-helper/common/utils/date.ts

```diff
@@ -15,6 +15,11 @@ const DATE_NOT_AVAILABLE_MESSAGE = __( 'Date N/A', 'wp-parsely' );
  */
 export function canProcessDate( date: Date|string ): boolean {
 	if ( 'string' === typeof date ) {
+		// Verify that date format is "YYYY-MM-DD".
+		if ( ! /^\d{4}-\d{2}-\d{2}$/.test( date ) ) {
```

Comment:
What will happen if the passed date is in `YYYY-DD-MM` format i.e. it will pass this condition but will there be any unexpected behavior?

Reply:
You've got a point.

There are regexes that are much more thorough (and very complicated) but I think that the real fix here would be to remove the capability of passing the date as a string.

The only function that accepts a date as a string is `removeDaysFromDate()`, so the fix might not be that complicated. This would be outside of the scope of this PR though, so noting this to look into it as a separate item.

Reply:
`canProcessDate()` no longer accepts a string as an argument and doesn't do any type of regex, so closing this comment as resolved.